### ISSUE TITLE
Evalution error with BatchNorm

### ISSUE THREAD

**Issue by Leong1230** [DLR-RM/stable-baselines3#614]
Hi! I am still faced with these problems since I use BatchNorm in my CustomCombinedExtractor. I have tried to modify some of the files with self.policy.eval(), but I still got unexpected test results. 
Are these problems fixed or not? I have reinstalled the codes under 'master', but it doesn't work. (The codes seem not changed as in the 'https://github.com/DLR-RM/stable-baselines3/pull/537/files')
Could you please offer some help? Thank you!

_Originally posted by @Leong1230 in https://github.com/DLR-RM/stable-baselines3/issues/537#issuecomment-945091297_

**Comment by Demetrio92** [comment]
I can reproduce this. @araffin 

```python
import gym
from stable_baselines3 import SAC
from stable_baselines3.common.evaluation import evaluate_policy
from stable_baselines3.common.vec_env import DummyVecEnv
from stable_baselines3.common.vec_env import VecFrameStack, VecMonitor

# params
N_sim_eval = 1000
gym_name = 'Pendulum-v0'
N_steps_train = 100000
n_stack = 2
log_dir = './Out/Logs'


print('')
print('---- Training ----')
print('')
env = DummyVecEnv([lambda: gym.make(gym_name)])
env = VecFrameStack(env, n_stack=n_stack)
env = VecMonitor(env, log_dir)
model = SAC(
    'MlpPolicy',
    env, verbose=1,
    learning_rate=1e-4,
)
model.learn(total_timesteps=N_steps_train)
model.save('Out/save_test')


print('')
print('---- Reward After model.load() ----')
print('')
model = SAC(
    'MlpPolicy',
    env, verbose=1,
    learning_rate=1e-4,
)
model.load('Out/save_test')
mean_reward, std_reward = evaluate_policy(model, env, n_eval_episodes=N_sim_eval, deterministic=True)
print(f'mean_reward:{mean_reward:.2f} +/- {std_reward:.2f}')


print('')
print('---- Reward After SAC.load() ----')
print('')
model = SAC.load('Out/save_test')
mean_reward, std_reward = evaluate_policy(model, env, n_eval_episodes=N_sim_eval, deterministic=True)
print(f'mean_reward:{mean_reward:.2f} +/- {std_reward:.2f}')
```
```
---- Reward After model.load() ----
mean_reward:-1394.89 +/- 214.19
---- Reward After SAC.load() ----
mean_reward:-144.10 +/- 80.72
```

**Comment by araffin** [comment]
> I can reproduce this. @araffin

See my answer in https://github.com/DLR-RM/stable-baselines3/issues/683#issuecomment-996213446 .
`model.load()` is not in-place, so you must use `model = SAC.load()` (we only show that in the doc, and if you use the RL Zoo, it is done automatically for you).
